sensor/common/processsignal/pipeline.go

```diff
@@ -30,15 +32,16 @@ type Pipeline struct {
 	enricher           *enricher
 	processFilter      filter.Filter
 	detector           detector.Detector
-	centralReady       concurrency.Signal
-	ctxCancel          context.CancelFunc
 	cm                 *channelmultiplexer.ChannelMultiplexer[*storage.ProcessIndicator]
+	ctxMux             *sync.Mutex
+	ctx                context.Context
+	ctxCancel          context.CancelCauseFunc
 }
 
 // NewProcessPipeline defines how to process a ProcessIndicator
 func NewProcessPipeline(indicators chan *message.ExpiringMessage, clusterEntities *clusterentities.Store, processFilter filter.Filter, detector detector.Detector) *Pipeline {
 	log.Debug("Calling NewProcessPipeline")
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancelCause(context.Background())
```

Comment:
🤩 I didn't know about this! We should add in the other components as well.

Reply:
It is a new thing and I wanted to give it a try: https://josephwoodward.co.uk/2023/01/context-cancellation-cause-with-cancel-cause